crates/driver/src/blockchain/mod.rs

```diff
@@ -55,25 +46,32 @@ impl Ethereum {
     }
 }
 
-// TODO This could probably do some caching, I might want to open an issue for
-// this?
 pub struct Contracts<'a>(&'a Ethereum);
 
 impl Contracts<'_> {
     /// The settlement contract.
-    pub async fn settlement(&self) -> Result<eth::Contract, Error> {
-        Ok(contracts::GPv2Settlement::deployed(&self.0.web3)
-            .await?
-            .address()
-            .into())
+    pub fn settlement(&self) -> contracts::GPv2Settlement {
+        let address = contracts::GPv2Settlement::raw_contract()
+            .networks
+            .get(&self.0.network.0)
```

Comment:
Note that, for historical reasons, the `networks` is a `HashMap<NetworkVersion, Deployment>` and **not** a `HashMap<NetworkName, Deployment>`. specifically, it indexes contracts based on the result of the RPC call `net_version`, which is typically a decimal number string.

In fact, for all networks we support, `net_version == to_string(eth_chainId)`.